File: src/main/java/com/flexingstudios/FlexingNetwork/api/menu/InvMenuImpl.java
package com.flexingstudios.FlexingNetwork.api.menu;

import com.flexingstudios.FlexingNetwork.api.util.Invs;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;

public abstract class InvMenuImpl extends Invs implements InvMenu {
    private final Inventory inventory;

    public InvMenuImpl(Inventory inventory) {
        this.inventory = inventory;
    }

    @Override
    public Inventory getInventory() {
        return inventory;
    }

    public void fill(int start, int end, ItemStack item) {
        for (int i = start; i < end; i++) {
            inventory.setItem(i, item == null ? null : item.clone());
        }
    }
}

File: src/main/kotlin/com/flexingstudios/flexingnetwork/api/menu/InvMenuImpl.kt
package com.flexingstudios.flexingnetwork.api.menu

import com.flexingstudios.flexingnetwork.api.player.NetworkPlayer
import com.flexingstudios.flexingnetwork.api.util.ClickType
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.ItemStack

 class InvMenuImpl(override val inventory: Inventory) : InvMenuKt {

    override fun addButton(slot: Int, item: ItemStack) {
        TODO("Not yet implemented")
    }

    override fun addButton(item: ItemStack, slot: Int) {
        TODO("Not yet implemented")
    }

    override fun onClick(item: ItemStack, player: NetworkPlayer, slot: Int, clickType: ClickType) {
        TODO("Not yet implemented")
    }

}